### PR TITLE
device: Added support for Calex smart wall switch

### PR DESCRIPTION
**Pull request by svrooij**
No battery level just yet.

### CODE REVIEW

**Comment by svrooij** [comment]
@Koenkk the tests did succeed on my machine and seems to be failing in code I didn't touch.

```plain
~/Projects/zigbee-herdsman-converters (device/calex-smart-switch ✘)✹ ᐅ npm run test      

> zigbee-herdsman-converters@14.0.5 test /Users/stephan/Projects/zigbee-herdsman-converters
> jest test

 PASS  test/otaCommon.test.js
 PASS  test/index.test.js (12.622 s)

Test Suites: 2 passed, 2 total
Tests:       23 passed, 23 total
Snapshots:   0 total
Time:        13.56 s, estimated 14 s
Ran all test suites matching /test/i.
~/Projects/zigbee-herdsman-converters (device/calex-smart-switch ✔) ᐅ npm run lint      

> zigbee-herdsman-converters@14.0.5 lint /Users/stephan/Projects/zigbee-herdsman-converters
> eslint .

```

File: devices.js [hunk 1]
```diff
@@ -9073,6 +9073,22 @@ const devices = [
...
+        toZigbee: [],
+        fromZigbee: [fz.command_off, fz.command_on, fz.command_step, fz.command_move_to_color_temp,
+            fz.command_move, fz.command_stop, fz.command_ehanced_move_to_hue_and_saturation,
+        ],
+        exposes: [e.action([
+            'on', 'off', 'color_temperature_move', 'brightness_step_up', 'brightness_step_down',
+            'brightness_move_up', 'brightness_move_down', 'brightness_stop',
+            'enhanced_move_to_hue_and_saturation',
+        ])],
+        meta: {configureKey: 1, disableActionGroup: true},
```
Comment (Koenkk) [inline devices.js:9090]: Battery shouldn't be difficult to support, add:

```js
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
            await reporting.batteryPercentageRemaining(endpoint);
        },
```

and add `e.battery()` to `exposes`.

Next start z2m, use the frontend and click the reconfigure button (or via MQTT: https://www.zigbee2mqtt.io/information/mqtt_topics_and_message_structure.html#zigbee2mqttbridgerequestdeviceconfigure). Check logs if OK, if you get timeout errors wakeup the device right before doing the configure. Next wait max 24 hours till battery reports.

Comment (svrooij) [reply devices.js:9090]: I don't think it has that endpoint. If I'm correct the `genPowerCfg` should also be in the endpoints right?

image

It also failed to bind that endpoint:
```plain
zigbee_1   | Zigbee2MQTT:error 2021-01-03 17:27:35: Failed to configure 'boven/werkkamer/kleur-schakelaar', attempt 3 (Error: Bind 0x00158d00024f3b6a/1 genPowerCfg from '0x00124b001caaac3d/1' failed (AREQ - ZDO - bindRsp after 10000ms)
zigbee_1   |     at Timeout._onTimeout (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:46:35)
zigbee_1   |     at listOnTimeout (internal/timers.js:554:17)
zigbee_1   |     at processTimers (internal/timers.js:497:7))
```

Comment (Koenkk) [reply devices.js:9090]: Looks that it doesn't support it indeed.

**Comment by Koenkk** [comment]
Thanks!

**Comment by svrooij** [comment]
@Koenkk jij bedankt!!! Zigbee2mqtt is awesome!

**Pull request merged**